docs/release_notes.rst

```diff
@@ -0,0 +1,207 @@
+=======================================================================================
+PUDL Release Notes
+=======================================================================================
+
+---------------------------------------------------------------------------------------
+0.4.0 (2021-07-XX)
+---------------------------------------------------------------------------------------
+This is a ridiculously large update including more than a year and a half's
+worth of work. We're going to start doing much more frequent incremental
+software releases with immediately usable processed data outputs as we move
+toward automating the release process in the coming months.
+
+New Data Coverage
+^^^^^^^^^^^^^^^^^
+
+* :doc:`data_sources/eia860` for 2004-2008 + 2019, plus eia860m through 2020.
+* :doc:`data_sources/eia923` for 2001-2008 + 2019
+* :doc:`data_sources/epacems` for 2019-2020
+* :doc:`data_sources/ferc1` for 2019
+* :ref:`US Census Demographic Profile (DP1) <data-censusdp1tract>` for 2010
+* :ref:`data-ferc714` for 2006-2019 (experimental)
+* :ref:`data-eia861` for 2001-2019 (experimental)
+
+Documentation & Data Accessibility
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+We've updated and (hopefully) clarified the documentation, and no longer expect
+most users to perform the data processing on their own. Instead, we are offering
+several methods of directly accessing already processed data:
+
+* Processed data archives on Zenodo that include a Docker container preserving
+  the required software environment for working with the data.
+* `A repository of PUDL example notebooks <https://github.com/catalyst-cooperative/pudl-examples>`__
+* `A JupyterHub instance <https://catalyst-cooperative.pilot.2i2c.cloud/>`__
+  hosted in collaboration with `2i2c <https://2i2c.org>`__
+* Browsable database access via `Datasette <https://datasette.io>`__ at
+  https://data.catalyst.coop
+
+Users who still want to run the ETL themselves will need to set up the
+:doc:`set up the PUDL development environment <dev/dev_setup>`
+
+Data Cleaning & Integration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* We now inject placeholder utilities in the cloned FERC Form 1 database when
+  respondent IDs appear in the data tables, but not in the respondent table.
+  This addresses a bunch of unsatisfied foreign key constraints in the original
+  databases published by FERC.
+* We're doing much more software testing and data validation, and so hopefully
+  we're catching more issues early on.
```

Comment:
should we add the (experimental!) net generation allocation?

Reply:
Please feel free to add a blurb explaining it!

Reply:
okay I added a section. I mostly stole content from the main module docs. Feel free to edit or condense